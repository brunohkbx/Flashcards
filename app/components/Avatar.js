import { View } from 'react-native';
import styled from 'styled-components';

const Avatar = styled(View)`
  width: ${props => props.width};
  height: ${props => props.height};
  borderRadius: ${props => props.borderRadius};
  backgroundColor: ${props => props.backgroundColor};
  justify-content: center;
  align-items: center;
`;

export default Avatar;
