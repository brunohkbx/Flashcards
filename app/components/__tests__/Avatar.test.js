import React from 'react';
import { Colors } from 'react-native-paper';
import { shallow } from 'enzyme';
import Avatar from '../Avatar';

describe('Avatar', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      width: 100,
      height: 100,
      borderRadius: 50,
      backgroundColor: Colors.white
    }, propOverrides);

    const wrapper = shallow(<Avatar {...defaultProps} />);

    return { wrapper };
  };

  it('renders properly', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
