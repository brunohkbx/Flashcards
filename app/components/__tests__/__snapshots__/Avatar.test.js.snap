// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders properly 1`] = `
<Component
  backgroundColor="#ffffff"
  borderRadius={50}
  className="sc-bdVaJa dixGsI"
  height={100}
  width={100}
/>
`;
